metric/emitter/prometheus.go

```diff
@@ -0,0 +1,188 @@
+package emitter
+
+import (
+	"fmt"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc/db"
+	"github.com/concourse/atc/metric"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type PrometheusEmitter struct {
+	buildsStarted     prometheus.Counter
+	buildsFinished    prometheus.Counter
+	buildsSucceeded   prometheus.Counter
+	buildsErrored     prometheus.Counter
+	buildsFailed      prometheus.Counter
+	buildsAborted     prometheus.Counter
+	buildsFinishedVec *prometheus.CounterVec
+	buildDurationsVec *prometheus.HistogramVec
+}
+
+type PrometheusConfig struct {
+	Address string `long:"prometheus-address" description:"Address to listen on to expose Prometheus metrics."`
```

Comment:
This should be two different fields: `BindIP` and `BindPort`, for consistency:

https://github.com/concourse/atc/blob/master/atccmd/command.go#L76-L77

Reply:
Sure!